### ISSUE TITLE
inconsistent bar gaps for time scale 

### ISSUE THREAD

**Issue by LukaszWiktor** [dc-js/dc.js#1136]
I have a daily data and I'm grouping by months. Some gaps between months have various sizes even if I set it explicitly using `.gap(1)`:

![image](https://cloud.githubusercontent.com/assets/2964543/15099303/08530424-1552-11e6-9a3e-b23653e70d48.png)

You can find code that renders the above chart here: https://jsfiddle.net/LukaszWiktor/Lwujrz7q/

My issue may be related with https://github.com/dc-js/dc.js/issues/952 but it may also have another cause. I guess it has something to do with different numer of days in each month (notice that February bars overlap March).

**Comment by gordonwoodhull** [comment]
Yes, I think you have diagnosed this correctly. I think to get this right, the width of a bar would have to be based on the position of the next bar.

Also related: PR #599, although I don't think it's a very general solution.

Best workaround I know is to coerce the dates to ordinal values, because widths are calculated correctly for ordinal scales. Totally a pain though.
